### ISSUE TITLE
Allow specifying remote canister module and candid interface

### ISSUE THREAD

**Issue by nmattia** [dfinity/sdk#2085]
It would be great if we could specify remote canisters in `dfx.json`, e.g.:

``` json
{
  "canisters": {
    "internet_identity": {
      "type": "remote",
      "candid": "https://github.com/dfinity/internet-identity/releases/latest/download/internet_identity_test.wasm",
      "wasm": "https://github.com/dfinity/internet-identity/releases/latest/download/internet_identity.did"
    }
  },...
}
```

This would be especially useful for widely used dapps like II that people might need to integrate with and run locally.

---

NOTE: my current workaround is to simply download the canister in the build (with `"build": "curl https://... -o my_canister.wasm"`) but that involves downloading on every single build, which isn't ideal.

**Comment by nmattia** [comment]
Now that [remote canisters](https://github.com/dfinity/sdk/blob/master/CHANGELOG.adoc#feat-remote-canister-support) are a thing, is there any chance we can make this happen?

Right now awkward workarounds are needed in pretty much every single app that needs to talk to existing ("remote") canisters: https://github.com/peterpeterparker/create-ic/commit/bba9e56980065986c75afc64d21796b61443708d#diff-4f5b61c8fb038c655795b31505582e5f687b4acd4b03a977e231669502eb9205R32-R35

**Comment by sesi200** [comment]
Implemented by #2563 and #2576
